dali/base/dadfs.cpp

```diff
@@ -5882,6 +5900,28 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
             throw MakeStringException(-1,"addSubFile: File %s is already a subfile of %s", sub->queryLogicalName(),queryLogicalName());
     }
 
+    void validate()
+    {
+        IDistributedFileDirectory &dfd = queryDistributedFileDirectory();
+        unsigned numSubfiles = root->getPropInt("@numsubfiles",0);
+        if (numSubfiles)
+        {
+            Owned<IPropertyTreeIterator> treeIter = root->getElements("SubFile");
+            ForEach(*treeIter)
+            {
+                IPropertyTree & st = treeIter->query();
+                StringBuffer subfilename;
+                st.getProp("@name", subfilename);
+                if (!dfd.exists(subfilename.str(), NULL))
+                    throw MakeStringException(-1, "Logical subfile '%s' doesn't exists!", subfilename.str());
+
+                if (!dfd.isSuperFile(subfilename.str(), NULL))
+                    if (!dfd.existsPhysical(subfilename.str(), NULL))
+                        throw MakeStringException(-1, "Physical Parts doesn't exists for logical file %s!", subfilename.str());
+            }
```

Comment:
May be worth also validating that there number of "SubFile"'s equals numSubFiles here.